src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -1028,6 +1028,20 @@ impl Step for PlainSourceTarball {
             builder.create(&cargo_config_dir.join("config.toml"), &config);
         }
 
+        // Delete extraneous directories
```

Comment:
So this will delete folders from my machine? That seems potentially dangerous?

Reply:
While it does delete directories, they are stored under `dist/tmp/tarball`, which is a temporary directory where bootstrap prepares the source directory for packaging. So it should be fine to delete stuff under this directory.